File: backend/services/email.js
const nodemailer = require('nodemailer');
const { ownerEmailTemplate } = require('../templates/ownerEmail');
const { applicantEmailTemplate } = require('../templates/applicantEmail');

// ─── Transporter factory ───────────────────────────────────────────────────────
// Uses Gmail SMTP with an App Password — no OAuth complexity needed
const createTransporter = () =>
  nodemailer.createTransport({
    service: 'gmail',
    auth: {
      user: process.env.EMAIL_FROM,
      pass: process.env.EMAIL_APP_PASSWORD,
    },
  });

// ─── Owner notification ────────────────────────────────────────────────────────
// Sends a structured application email to the Consulting Group inbox
const sendOwnerEmail = async (data) => {
  const transporter = createTransporter();

  const attachments = [];
  if (data.resumeBuffer && data.resumeFileName) {
    attachments.push({
      filename: data.resumeFileName,
      content:  data.resumeBuffer,
    });
  }

  const info = await transporter.sendMail({
    from:        `"Consulting Group Agent" <${process.env.EMAIL_FROM}>`,
    to:          process.env.OWNER_EMAIL,
    subject:     `New Application — ${data.jobRole} | ${data.location} | ${data.shift}`,
    html:        ownerEmailTemplate(data),
    attachments,
  });

  console.log(`[Email] Owner notified — MessageId: ${info.messageId}`);
  return info;
};

// ─── Applicant confirmation ────────────────────────────────────────────────────
// Sends a friendly confirmation receipt to the job seeker
const sendApplicantEmail = async (data) => {
  const transporter = createTransporter();

  const info = await transporter.sendMail({
    from:    `"Consulting Group" <${process.env.EMAIL_FROM}>`,
    to:      data.email,
    subject: 'Your Application to Consulting Group — Received',
    html:    applicantEmailTemplate(data),
  });

  console.log(`[Email] Applicant confirmed — MessageId: ${info.messageId}`);
  return info;
};

module.exports = { sendOwnerEmail, sendApplicantEmail };

File: backend_py/services/email.py
import os
import smtplib
import asyncio
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
from email.mime.base import MIMEBase
from email import encoders

from templates.owner_email import owner_email_template
from templates.applicant_email import applicant_email_template


# ─── Low-level sender ─────────────────────────────────────────────────────────
def _send_smtp(msg: MIMEMultipart):
    with smtplib.SMTP_SSL("smtp.gmail.com", 465) as server:
        server.login(os.getenv("EMAIL_FROM"), os.getenv("EMAIL_APP_PASSWORD"))
        server.send_message(msg)


# ─── Owner notification ───────────────────────────────────────────────────────
# Sends structured application data + resume attachment to the owner inbox
async def send_owner_email(data: dict):
    msg = MIMEMultipart()
    msg["From"]    = f"Consulting Group Agent <{os.getenv('EMAIL_FROM')}>"
    msg["To"]      = os.getenv("OWNER_EMAIL")
    msg["Subject"] = f"New Application — {data.get('jobRole')} | {data.get('location')} | {data.get('shift')}"

    msg.attach(MIMEText(owner_email_template(data), "html"))

    # Attach resume if uploaded
    if data.get("resumeBuffer") and data.get("resumeFileName"):
        part = MIMEBase("application", "octet-stream")
        part.set_payload(data["resumeBuffer"])
        encoders.encode_base64(part)
        part.add_header("Content-Disposition", f'attachment; filename="{data["resumeFileName"]}"')
        msg.attach(part)

    # Run blocking SMTP in thread pool — don't block the event loop
    await asyncio.to_thread(_send_smtp, msg)
    print(f"[Email] Owner notified — {data.get('name')} | {data.get('jobRole')}")


# ─── Applicant confirmation ───────────────────────────────────────────────────
# Sends branded confirmation receipt to the job seeker
async def send_applicant_email(data: dict):
    msg = MIMEMultipart()
    msg["From"]    = f"Consulting Group <{os.getenv('EMAIL_FROM')}>"
    msg["To"]      = data["email"]
    msg["Subject"] = "Your Application to Consulting Group — Received"

    msg.attach(MIMEText(applicant_email_template(data), "html"))

    await asyncio.to_thread(_send_smtp, msg)
    print(f"[Email] Applicant confirmed — {data.get('email')}")
